instructor/common.py

```diff
@@ -0,0 +1,604 @@
+from django.shortcuts import render_to_response, redirect
+from django.template import RequestContext
+from django import forms
+from django.forms.models import modelformset_factory, inlineformset_factory
+from django.forms.models import modelform_factory
+from instructor import models
+from instructor import compiler
+from django.contrib.auth.models import User
+import datetime
+
+from backend.models import Assignment, StudentAssignment
+
+
+def courses(request):
+    user = request.user
+    message = ''
+    CourseFormSet = modelformset_factory(models.Course, extra=1, fields=['name', 'code'], can_delete=True)
+    if request.method == "POST":
+        formset = CourseFormSet(request.POST)
+        if ( formset.is_valid()):
+            message = "Thank you"
+            entries = formset.save(commit=False)
+            for form in entries:
+                form.owner = request.user
+                form.save()
+        else:
+            message = "Something went wrong"
+
+    return render_to_response('instructor/courses_new.html',
+                              {'formset': CourseFormSet(queryset=models.Course.objects.filter(owner=user)),
+                               'message': message},
+                              context_instance=RequestContext(request))
+
+
+def course_delete(request, pk):
+    print "DELETE {}".format(pk)
+    models.Course.objects.get(id=pk).delete()
+    return redirect('common_course')
+
+
+def get_account_type(request):
+    account_type = ''
+    if request.user.id and len(request.user.groups.all()) > 0:
+        account_type = request.user.groups.all()[0].name
+    return account_type
+
+
+def assignments(request):
+
+    account_type = get_account_type(request)
+    public_list = models.Assignment.objects.filter(access='Public')
+
+    return render_to_response('instructor/assignments.html',
+                              {'assignments': public_list},
+                              context_instance=RequestContext(request))
+
+
+def assignments_delete(request, pk):
+    models.Assignment.objects.get(id=pk).delete()
+    return redirect('common_assignments')
+
+def create_new_assignment(request):
+    request.session['new'] = True
+    request.session['assignment_name'] = ''
+    request.session['based_on'] = ''
+    return redirect('common_assignment_setup')
+
+
+def assignment_setup(request):
+    if request.method == "POST":
+        #Have to give error for empty name
+        request.session['assignment_name'] = request.POST['name']
+        if request.POST['based_on']:
+            request.session['based_on'] = request.POST['based_on']
+
+        if 'continue' in request.POST:
+            return redirect('common_course_setup')
+
+    all_assignments = models.Assignment.objects.all()
```

Comment:
this is another permission question, do we want to just add a user attribute to the assignments and courses, and filter on that, or do we want to use groups, or just let it all hang out?